### PR TITLE
Add prometheus

### PR DESCRIPTION
**Pull request by apoorvsadana**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -27,10 +27,11 @@
...
 broadcast
+.idea
 
 # ignore genesis.json
 .hive
 
 artifacts/
 build/
-.kakarot/
+.kakarot/
```
Comment (greged93) [inline .gitignore:37]: can you just add a line break here?

Comment (apoorvsadana) [reply .gitignore]: after kakarot?

Comment (greged93) [reply .gitignore]: yes

Comment (apoorvsadana) [reply .gitignore]: fixed

File: src/eth_rpc/mod.rs [hunk 2]
```diff
@@ -39,15 +49,33 @@ pub async fn run_server(
...
+        let _ = init_prometheus(
+            SocketAddr::new(
+                std::net::IpAddr::V4(Ipv4Addr::new(0, 0, 0, 0)),
+                get_env_or_default("PROMETHEUS_PORT", "9615").parse().unwrap(),
+            ),
+            registry.clone(),
+        )
+        .await;
+    });
+    let rpc_middleware = RpcServiceBuilder::new().option_layer(metrics);
```
Comment (greged93) [inline src/eth_rpc/mod.rs:64]: Can you explain for me what you are doing here? Also is it ok to clone the registry or should the init_prometheus get a reference to it?

File: src/prometheus_handler/mod.rs [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+}
+
+/// Init prometheus using the given listener.
+async fn init_prometheus_with_listener(listener: tokio::net::TcpListener, registry: Registry) -> Result<(), Error> {
+    let listener = hyper::server::conn::AddrIncoming::from_listener(listener)?;
+    log::info!("〽️ Prometheus exporter started at {}", listener.local_addr());
+
+    let service = make_service_fn(move |_| {
+        let registry = registry.clone();
+
```
Comment (greged93) [inline src/prometheus_handler/mod.rs:95]: do you need to clone it twice?

**Review by greged93 — COMMENTED** [review]
small comments, mainly for my understanding

File: src/eth_rpc/mod.rs (hunk 2 above)
Comment (apoorvsadana) [reply src/eth_rpc/mod.rs]: Added comments. Good catch, it works without the clone!

**Review by greged93 — APPROVED** [review]
lgtm!

**Pull request merged**